### PR TITLE
API sends an invite to two Participants

### PR DESCRIPTION
**Pull request by tochman**
PT: https://www.pivotaltracker.com/story/show/180227722

## User Story 
```
As an API
In order to seamlessly invite organization members to a Fika
I would like to send invites directly to participants' google calendars
```

### Gems installed
- google-api-client
- lockbox
- httplog

## Tasks
- Add `webmock` gem for testing `google-api-client` calls for authentication and event creation
- Create send invitation spec and fixture files
- Create `invite_service` in `app/services`
- Update fika#model with validations and call `invite_service`
- Update Fika#create spec with new sad paths
- Create an robust error handler for google calendar

### Screenshots
image

### CODE REVIEW

File: app/models/fika.rb [hunk 1]
```diff
@@ -6,13 +6,11 @@ class Fika < ApplicationRecord
...
-    participants = Participant.all
-    possible_participant_pairings = participants.to_a.permutation(2).to_a
+    possible_participant_pairings = Participant.all.to_a.permutation(2).to_a
     unique_pairings = []
     possible_participant_pairings.each do |pair|
       unique_pairings.push(pair.sort_by(&:id))
     end
-    unique_pairings.uniq!
-    unique_pairings
+    unique_pairings.uniq
```
Comment (dorianbuck) [inline app/models/fika.rb:14]: Is this change due to formating or refactoring?

**Review by dorianbuck — APPROVED** [review]
Awesome!

File: app/models/fika.rb (hunk 1 above)
Comment (tochman) [reply app/models/fika.rb:14]: refactoring

Comment (maxarvid) [reply app/models/fika.rb:14]: This is clever and I totally missed it. When calling `.uniq!` on an array, the expression itself returns the new array if changes are made, but `nil` if all the elements of the array are the same. That is why we needed to return the array by itself on a separate line when using `.uniq!`. We fails some of our tests if this instance method returns `nil`, but the problem is solved by using the regular `.uniq`, which will always return a new array with any duplicates filtered out.

Comment (tochman) [reply app/models/fika.rb:14]: 👍

File: spec/fixtures/files/bad_request_response.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                "reason": "timeRangeEmpty",
+                "message": "The specified time range is empty.",
+                "locationType": "parameter",
+                "location": "timeMax"
+            }
+        ],
+        "code": 400,
+        "message": "The specified time range is empty."
+    }
+}
```
Comment (tochman) [inline spec/fixtures/files/bad_request_response.json]: Formatting is off. Use 2 spaces

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -1,6 +1,10 @@
+require 'webmock/rspec'
 require 'spec_helper'
 require 'simplecov'
 SimpleCov.start
+# WebMock.enable!
+
+# WebMock.disable_net_connect!(allow: 'www.googleapis.com/oauth2/v4/token')
```
Comment (tochman) [inline spec/rails_helper.rb]: Please remove comments

**Review by dorianbuck — APPROVED** [review]

**Review by AugustEnhager — APPROVED** [review]

**Pull request merged**